### PR TITLE
Update CHANGES.txt for 2.1.0 release.

### PR DESCRIPTION
**Pull request by daspecster**
Per #573, 

I'm not exactly sure what exactly is important to you to have in the changelog.

I ran `git log 2.0.0..HEAD --oneline` on `master` and then trimmed up the following output, based on what sounded important/useful.

```
1391c7e Update CHANGES.txt for 2.1.0 release.
1e71012 Merge pull request #562 from sigmavirus24/bug/561
d4b8f49 Stop checking for string option type
0babee5 Merge pull request #557 from gitter-badger/gitter-badge
df8a6c0 Add Gitter badge
001a0ce Merge pull request #556 from markpeek/markpeek-async-def
72acd6f Report E302 for blank lines before an "async def"
4438622 Updated tox to do same tests as Travis CI
5c4c081 Removed test code unrelated to utf-8 checks
08f4891 Merge branch 'gh-536'
a0d5d5d Added test case based on @gvanrossum comment
2a5e575 Updated checker variable based on conversation in #536
10f4e51 Added extra Okay example to test against
f55b1c5 Fixed placement of changelog entry
6b75c4d Fix issue #400: Require two blank lines after toplevel def, class
dc08dad Updated changelog with 2.1 milestone changes
5a6ffd5 Updated rename note in README
e01ca2f Updated to use sheilds.io badges
15b1b42 Updated text about running tests with tox; Closes #552
f8af83d Merge pull request #539 from sigmavirus24/max-line-length-noqa
1fcd279 Merge pull request #540 from sigmavirus24/remove-warnings
95ed849 Remove unused imports and assignments
7fa71af Update maximum_line_length to use Checker.noqa
6cb59d6 Updated changelog to specify changes in 2.1.0
4ccdc55 Merge pull request #525 from mtmiller/issue341
60859d6 Updated version number of next release to 2.1.0
e15075a Updated Travis CI doc link
d2171cd Updated docs related to tested Python versions
42c8987 Added passing test case related to #376
ca7e43b Fixed wrapping of text in developer.rst
fc8c799 Added skip missing interpreters flag to tox config
45503fc Updated developer documentation for testing
5be92ad Add test-requirements.txt with basic test packages
7ffd648 Reverted a few spelling changes
adcb3a2 Fix typos
e8d358e Re-ordered package imports alphabetically
64edd89 Bumped version number post release
c122532 Merge pull request #532 from mtmiller/doc-update
77f5543 Merge pull request #530 from iamaspacecow/pycodestyle-name-change
26c64ba Change all references to the pep8 project to say pycodestyle This fixes issue #518, pep8 still referenced in the cli help command As a side effect, `[pep8]` in setup.cfg now becomes `[pycodestyle]` Also, changed the path for the config file from ~/.config/pep8 to ~/.config/pycodestyle These feel like changes that should have come with the jump to version 2.0.0, as they are breaking, but support for  as a name can still be added if it's desired enough
b98ed7b Update Python version list in docs
fdf3e6a Report E742 and E743 for badly named functions and classes
017d3b5 Also report E741 on 'global' and 'nonlocal' statements
4c8152c Add more doctests for E741
b02d768 Merge pull request #522 from arthall/issue-314
26893da Report E741 for prohibited single-letter variables
a99bcec Never use 'l' (lowercase letter el) as a variable name
5949086 Modified patch for issue #314 so tests in E73.py pass
3944d9e Merge pull request #520 from kjcole/master
6fe11c4 - Added venv/ (virtualenv) to list of git ignored filename patterns
5cf3c7a Merge latest master into issue-314
bbefb19 Fixing compound_statement not to be quadratic in # of :s
```

LMKWYT and I'll update accordingly.

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Change all references to the pep8 project to say pycodestyle; #530
+
 Changes:
 
-* Added check E74x for using variables named 'l', 'O', or 'I'; #341
-* Improved performance of `compound_statements` check; #314 / #522
-* Fixed remaining references to `pep8`; #518 / #530
-* Added `noqa` support for `maximum_line_length` check; #538
-* Added check E305 for two blank lines after toplevel methods and classes; #400
+* Add Gitter badge; #557
```
Comment (sigmavirus24) [inline CHANGES.txt]: I don't think this is necessary to have here.

File: CHANGES.txt [hunk 2]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Report E302 for blank lines before an "async def"; #556
+* Updated tox to do same tests as Travis CI (`setup.py test`).
```
Comment (sigmavirus24) [inline CHANGES.txt]: This can also be removed.

File: CHANGES.txt [hunk 3]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Updated to use sheilds.io badges.
```
Comment (sigmavirus24) [inline CHANGES.txt]: This too

File: CHANGES.txt [hunk 4]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Updated text about running tests with tox; #552
```
Comment (sigmavirus24) [inline CHANGES.txt]: This too

File: CHANGES.txt [hunk 5]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Update maximum_line_length to use Checker.noqa; 538
```
Comment (sigmavirus24) [inline CHANGES.txt]: This is an internal change to how things work. It shouldn't be noted here.

File: CHANGES.txt [hunk 6]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Updated Travis CI doc link.
```
Comment (sigmavirus24) [inline CHANGES.txt]: This shouldn't be of interest to users.

File: CHANGES.txt [hunk 7]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Updated docs related to tested Python versions.
```
Comment (sigmavirus24) [inline CHANGES.txt]: Perhaps "Update our list of tested and supported Python versions which are now ..."

File: CHANGES.txt [hunk 8]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Added skip missing interpreters flag to tox config.
```
Comment (sigmavirus24) [inline CHANGES.txt]: This doesn't affect end-users.

File: CHANGES.txt [hunk 9]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Add test-requirements.txt with basic test packages.
```
Comment (sigmavirus24) [inline CHANGES.txt]: This doesn't affect end-users either

File: CHANGES.txt [hunk 10]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Report E742 and E743 for badly named functions and classes.
```
Comment (sigmavirus24) [inline CHANGES.txt]: Can you expand this a bit? What's "badly named"?

File: CHANGES.txt [hunk 11]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Also report E741 on 'global' and 'nonlocal' statements.
+* Report E741 for prohibited single-letter variables.
```
Comment (sigmavirus24) [inline CHANGES.txt]: Consolidate this and the one above into one line-note, please.

File: CHANGES.txt [hunk 12]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Added venv/ (virtualenv) to list of git ignored filename patterns.
```
Comment (sigmavirus24) [inline CHANGES.txt]: This doesn't affect end-users. Please remove it.

File: CHANGES.txt [hunk 13]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+
+Bugs:
 
+* Stop checking for string option type; #561
```
Comment (sigmavirus24) [inline CHANGES.txt]: This is not helpful. Please summarize the bug a bit better.

File: CHANGES.txt [hunk 14]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Require two blank lines after toplevel def, class; #536
+* Badly named single-letter variable names are not detected; #341
```
Comment (sigmavirus24) [inline CHANGES.txt]: Is this related to the "badly named" line item above? If so, can we just have one line item in the appropriate place?

File: CHANGES.txt [hunk 15]
```diff
@@ -4,14 +4,33 @@ Changelog
...
+* Fixing compound_statement not to be quadratic in # of :s; #314
```
Comment (sigmavirus24) [inline CHANGES.txt]: Can we reword this a bit, e.g., 

```
Remove accidentally quadratic computation based on the number of colons. This will make pycodestyle faster in some cases.
```

File: CHANGES.txt (hunk 13 above)
Comment (daspecster) [inline CHANGES.txt]: How about "Fix opt_type AssertionError when using Flake8 2.6.2 and pycodestyle 2.0.0; #561"

**Comment by daspecster** [comment]
@sigmavirus24, sorry the slow turn around.

PTAL when you have a chance!

File: CHANGES.txt (hunk 13 above)
Comment (sigmavirus24) [inline CHANGES.txt]: "Prevent unhandled assertion error in option parsing with Flake8 2.6.2"?

File: CHANGES.txt [hunk 16]
```diff
@@ -4,14 +4,22 @@ Changelog
...
+
 Changes:
 
-* Added check E74x for using variables named 'l', 'O', or 'I'; #341
-* Improved performance of `compound_statements` check; #314 / #522
-* Fixed remaining references to `pep8`; #518 / #530
-* Added `noqa` support for `maximum_line_length` check; #538
-* Added check E305 for two blank lines after toplevel methods and classes; #400
+* Report E302 for blank lines before an "async def"; #556
+* Update our list of tested and supported Python versions which are 2.6 through 3.5 as well as the nightly Python build and PyPy.
```
Comment (sigmavirus24) [inline CHANGES.txt]: Do we test on 3.0, 3.1, and 3.2? I didn't think so

Comment (daspecster) [inline CHANGES.txt]: Sorry, I should have looked at Travis.
I just copied that from https://github.com/PyCQA/pycodestyle/commit/d2171cd3b6664b8afe8b51071b835de2c29c3c30#diff-d37c6b239400622b3c3d8738037e9bdd.

So that may need to be updated as well. I can add that to this PR or make a new one?

Comment (daspecster) [inline CHANGES.txt]: Also, it looks like python 3.2 is tested?

https://github.com/PyCQA/pycodestyle/blob/master/.travis.yml#L4-L9

Comment (sigmavirus24) [inline CHANGES.txt]: We should drop support for 3.2 and add a new PR to update our language to be far more specific about what versions we're testing.

Thanks @daspecster

**Comment by daspecster** [comment]
No problem!

**Comment by IanLee1521** [comment]
Thank you!

I'll work on getting the new release out today.

**Comment by richrd** [comment]
Hi! I don't want to pressure you or be annoying, so sorry about this. I know how busy life can be as a dev. However it would be super nice to get this released, since my build has been broken due to #561 for a few months (https://github.com/richrd/suplemon).

It's not really a big deal since my code is fine, but the red badge just looks bad. Other than that I really love the tool and It's been instrumental in maintaining my code quality. :heart:

**Comment by daspecster** [comment]
@richrd, there was a question about backwards compatibility that you can see in https://github.com/PyCQA/pycodestyle/issues/573.
